plugin/src/main/scala/se/nullable/sbtix/NixWriter.scala

```diff
@@ -86,17 +86,17 @@ case class NixArtifactCollection(artifacts: Seq[NixArtifact])  extends NixBuilde
 
   def toNixValue: String = 
       s"""{
-          |  ${indent(artifacts.map(_.toNix).mkString("\r\n"))}
+          |  ${indent(artifacts.distinct.map(_.toNix).mkString("\r\n"))}
           |}""".stripMargin
 
 }
 
-case class NixArtifact(repoName:String, relative: String, url: URL, sha256: String) extends NixBuilder {
-  def toNixRef = s"${quote(repoName + "/" + relative.stripPrefix("/"))}"
+case class NixArtifact(repoName:String, relative: String, url: String, sha256: String) extends NixBuilder {
```

Comment:
Does converting this earlier help with the deduplication? It'd be nice to keep the types meaningful for as long as possible.